+nosnoc/+solver/MpccSolver.m

```diff
@@ -72,16 +69,294 @@
             
             if isa(mpcc, 'vdx.problems.Mpcc') % We can properly interleave complementarities if we get a vdx.Mpcc
                 use_vdx = true;
-                % TODO(@anton) implement this
+                mpcc.finalize_assignments();
+                casadi_symbolic_mode = mpcc.w.casadi_type;
+                nlp = vdx.Problem('casadi_type', casadi_symbolic_mode);
+
+                % Copy over w, g, and p.
+                nlp.w = copy(mpcc.w);
+                nlp.w.problem = nlp;
+                nlp.g = copy(mpcc.g);
+                nlp.g.problem = nlp;
+                nlp.p = copy(mpcc.p);
+                nlp.p.problem = nlp;
+                nlp.f = mpcc.f;
+                g_vars = nlp.g.get_vars();
+                pre_sort_g_ind = cellfun(@(x) [x.indices{:}], g_vars, "UniformOutput", false);
+                pre_sort_g_ind = [pre_sort_g_ind{:}];
+                w_vars = nlp.w.get_vars();
+                pre_sort_w_ind = cellfun(@(x) [x.indices{:}], w_vars, "UniformOutput", false);
+                pre_sort_w_ind = [pre_sort_w_ind{:}];
+                p_vars = nlp.p.get_vars();
+                pre_sort_p_ind = cellfun(@(x) [x.indices{:}], p_vars, "UniformOutput", false);
+                pre_sort_p_ind = [pre_sort_p_ind{:}];
+
+                nlp.p.sigma_p = {{'sigma_p', 1}, opts.sigma_0};
+
+
+                % Create relaxation slacks/parameters
+                switch opts.homotopy_steering_strategy
+                  case HomotopySteeringStrategy.DIRECT
+                    % Nothing to be done here.
+                  case HomotopySteeringStrategy.ELL_INF
+                    % adding a scalar elastic variable to nlp.w which augments the original mpcc.w
+                    nlp.w.s_elastic = {{'s_elastic', 1}, opts.s_elastic_min, opts.s_elastic_max, opts.s_elastic_0};
+                    if opts.decreasing_s_elastic_upper_bound
+                        nlp.g.s_ub = {nlp.w.s_elastic() - nlp.p.sigma_p(), -inf, 0};
+                    end
+                    if opts.objective_scaling_direct
+                        nlp.f = nlp.f + (1/nlp.p.sigma_p())*nlp.w.s_elastic(); % penalize the elastic more and more with decreasing sigma_p
+                    else
+                        nlp.f = nlp.p.sigma_p()*nlp.f + nlp.w.s_elastic(); % reduce the weight of the initial objective with decreasing sigma_p
+                    end
+                  case HomotopySteeringStrategy.ELL_1
+                    % Ell 1 steering strategy slacks are added during creation of constraints
+                end
+
```

Comment:
No, often just carelessness brought on by my linter settings for other languages. I plan on using the side work I've done for `tree-sitter-matlab` to write a 3rd party linting tool for matlab (because frankly the mathworks one is :wastebasket: ) but time is never on ones side.